core/src/main/java/cucumber/runner/AbstractEventBus.java

```diff
@@ -4,52 +4,47 @@
 import cucumber.api.event.EventHandler;
 
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-public class DefaultEventBus implements EventBus {
-    private final TimeService stopWatch;
+abstract class AbstractEventBus implements EventBus {
```

Comment:
This keeps the handler registration stuff in a single place. Makes the other implementations cleaner.